code/game/machinery/Freezer.dm

```diff
@@ -104,59 +104,48 @@
-        // for a list of parameters and their descriptions see the code docs in \code\modules\nano\nanoui.dm
-		ui = new(user, src, ui_key, "freezer.tmpl", "Gas Cooling System", 540, 300)
-		// open the new ui window
+		ui = new(user, src, ui_key, "ThermoMachine", "Gas Cooling System", 540, 200)
 		ui.open()
-		// auto update every Master Controller tick
-		ui.set_auto_update(1)
-
-/obj/machinery/atmospherics/unary/cold_sink/freezer/ui_data(mob/user, ui_key = "main", datum/topic_state/state = GLOB.default_state)
-	var/data[0]
-	data["on"] = on ? 1 : 0
-	data["gasPressure"] = round(air_contents.return_pressure())
-	data["gasTemperature"] = round(air_contents.temperature)
-	data["gasTemperatureCelsius"] = round(air_contents.temperature - T0C,1)
+
+/obj/machinery/atmospherics/unary/cold_sink/freezer/tgui_data(mob/user)
+	var/list/data = list()
+	data["on"] = on
+	data["pressure"] = round(air_contents.return_pressure())
+	data["temperature"] = round(air_contents.temperature)
+	data["temperatureCelsius"] = round(air_contents.temperature - T0C,1)
 	if(air_contents.total_moles() == 0 && air_contents.temperature == 0)
-		data["gasTemperatureCelsius"] = 0
-	data["minGasTemperature"] = round(min_temperature)
-	data["maxGasTemperature"] = round(T20C)
-	data["targetGasTemperature"] = round(current_temperature)
-	data["targetGasTemperatureCelsius"] = round(current_temperature - T0C,1)
-
-	var/temp_class = "good"
-	if(air_contents.temperature > (T0C - 20))
-		temp_class = "bad"
-	else if(air_contents.temperature < (T0C - 20) && air_contents.temperature > (T0C - 100))
-		temp_class = "average"
-	data["gasTemperatureClass"] = temp_class
+		data["temperatureCelsius"] = 0
+	data["min"] = round(min_temperature)
+	data["max"] = round(T20C)
+	data["target"] = round(current_temperature)
+	data["targetCelsius"] = round(current_temperature - T0C,1)
 	return data
 
-/obj/machinery/atmospherics/unary/cold_sink/freezer/Topic(href, href_list)
+/obj/machinery/atmospherics/unary/cold_sink/freezer/tgui_act(action, params)
 	if(..())
-		return 1
-	if(href_list["toggleStatus"])
-		src.on = !src.on
-		update_icon()
-	else if(href_list["minimum"])
-		current_temperature = min_temperature
-	else if(href_list["maximum"])
-		current_temperature = T20C
-	else if(href_list["temp"])
-		var/amount = text2num(href_list["temp"])
-		if(amount > 0)
-			src.current_temperature = min(T20C, src.current_temperature+amount)
-		else
-			src.current_temperature = max(min_temperature, src.current_temperature+amount)
-	src.add_fingerprint(usr)
-	return 1
+		return
+	add_fingerprint(usr)
+
+	switch(action)
+		if("power")
+			on = !on
+			update_icon()
+		if("minimum")
+			current_temperature = min_temperature
+		if("maximum")
+			current_temperature = T20C
+		if("temp")
+			var/amount = params["temp"]
+			amount = text2num(amount)
+			if(amount > 0)
+				current_temperature = min(T20C, amount)
+			else
+				current_temperature = max(min_temperature, amount)
```

Comment:
```suggestion
			current_temperature = clamp(amount, min_temperature, T20C)
```
Something like this might be clearer. Unless you intend to have different boundaries in these two cases.